### ISSUE TITLE
C# Generated Code is not strongly typed

### ISSUE THREAD

**Issue by johnamcruz** [Azure/autorest#1890]
The swagger code below is not generating a strong type objects for the GET methods.

```JSON
{
  "swagger": "2.0",
  "info": {
    "version": "1.0.0",
    "title": "Swagger Flights (Simple)"
  },
  "host": "xxx.co",
  "basePath": "/1",
  "schemes": [
    "https"
  ],
  "paths": {
    "/flights": {
      "get": {
        "operationId": "findFlights",
        "consumes": [
          "application/json"
        ],
        "produces": [
          "application/json"
        ],
        "parameters": [
          {
            "name": "ArrivalAirportCode",
            "in": "query",
            "required": false,
            "type": "string"
          },
          {
            "name": "DepartureAirportCode",
            "in": "query",
            "required": false,
            "type": "string"
          },
          {
            "name": "DepartureDateTimeStart",
            "in": "query",
            "required": false,
            "type": "string"
          },
          {
            "name": "FlightNumber",
            "in": "query",
            "required": false,
            "type": "string"
          },
          {
            "name": "AvailableSeatType",
            "in": "query",
            "required": false,
            "type": "string"
          },
          {
            "name": "DepartureDateTimeEnd",
            "in": "query",
            "required": false,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "200 response",
            "schema": {
              "type": "array",
              "items": {
                "$ref": "#/definitions/flight"
              }              
            }
          },
          "400": {
            "description": "400 response",
            "schema": {
              "$ref": "#/definitions/errorModel"
            }
          }
        },
        "security": [
          {
            "sigv4": []
          },
          {
            "api_key": []
          }
        ]
      }
    },
    "/flights/{flightNumber}": {
      "get": {
        "operationId": "findFlightByFlightNumber",
        "consumes": [
          "application/json"
        ],
        "produces": [
          "application/json"
        ],
        "parameters": [
          {
            "name": "flightNumber",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "200 response",
            "schema": {
              "$ref": "#/definitions/flight"
            }
          },
          "400": {
            "description": "400 response",
            "schema": {
              "$ref": "#/definitions/errorModel"
            }
          },
          "404": {
            "description": "404 response",
            "$ref": "#/definitions/errorModel"
          }
        },
        "security": [
          {
            "sigv4": []
          },
          {
            "api_key": []
          }
        ]
      }
    },
    "/flights/{flightNumber}/Info": {
      "get": {
        "operationId": "findFlightInfoByFlightNumber",
        "consumes": [
          "application/json"
        ],
        "produces": [
          "application/json"
        ],
        "parameters": [
          {
            "name": "flightNumber",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "200 response",
            "schema": {
              "$ref": "#/definitions/FlightInformation"
            }
          },
          "400": {
            "description": "400 response",
            "schema": {
              "$ref": "#/definitions/errorModel"
            }
          }
        },
        "security": [
          {
            "sigv4": []
          },
          {
            "api_key": []
          }
        ]
      }
    },
    "/flights/{flightNumber}/Info/Equipment": {
      "get": {
        "operationId": "findFlightEquipmentInfoByFlightNumber",
        "consumes": [
          "application/json"
        ],
        "produces": [
          "application/json"
        ],
        "parameters": [
          {
            "name": "flightNumber",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "200 response",
            "schema": {
              "$ref": "#/definitions/FlightEquipment"
            }
          },
          "400": {
            "description": "400 response",
            "schema": {
              "$ref": "#/definitions/errorModel"
            }
          }
        },
        "security": [
          {
            "sigv4": []
          },
          {
            "api_key": []
          }
        ]
      }
    },
    "/flights/{flightNumber}/Info/Equipment/{EquipmentTypeId}": {
      "get": {
        "operationId": "findFlightEquipmentInfoByFlightNumberAndEquipmentTypeId",
        "consumes": [
          "application/json"
        ],
        "produces": [
          "application/json"
        ],
        "parameters": [
          {
            "name": "flightNumber",
            "in": "path",
            "required": true,
            "type": "string"
          },
          {
            "name": "EquipmentTypeId",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "200 response",
            "schema": {
              "$ref": "#/definitions/flight"
            }
          },
          "400": {
            "description": "400 response",
            "schema": {
              "$ref": "#/definitions/errorModel"
            }
          }
        },
        "security": [
          {
            "sigv4": []
          },
          {
            "api_key": []
          }
        ]
      }
    },
    "/flights/{flightNumber}/Status": {
      "get": {
        "operationId": "findFlightStatusByFlightNumber",
        "consumes": [
          "application/json"
        ],
        "produces": [
          "application/json"
        ],
        "parameters": [
          {
            "name": "flightNumber",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "200 response",
            "schema": {
              "$ref": "#/definitions/FlightStatus"
            }
          },
          "400": {
            "description": "400 response",
            "schema": {
              "$ref": "#/definitions/errorModel"
            }
          }
        },
        "security": [
          {
            "sigv4": []
          },
          {
            "api_key": []
          }
        ]
      }
    }   
  },
  "securityDefinitions": {
    "api_key": {
      "type": "apiKey",
      "name": "x-api-key",
      "in": "header"
    },
    "sigv4": {
      "type": "apiKey",
      "name": "Authorization",
      "in": "header",
      "x-amazon-apigateway-authtype": "awsSigv4"
    }
  },
  "definitions": {
    "flight": {
      "type": "object",
      "required": [
        "Number"
      ],
      "properties": {
        "Number": {
          "type": "integer",
          "format": "int64"
        },
        "AirlineCode": {
          "type": "string"
        },
        "AirlineName": {
          "type": "string"
        },
        "Date": {
          "type": "string"
        },
        "CodeSharing": {
          "type": "string"
        },
        "OperatingCarrier": {
          "type": "string"
        },
        "SecureFlightSpecified": {
          "type": "string"
        },
        "SmokingSpecified": {
          "type": "string"
        },
        "AircraftType": {
          "type": "string"
        },
        "FlightInformation": {
          "$ref": "#/definitions/FlightInformation"
        },
        "FlightStatus": {
          "$ref": "#/definitions/FlightStatus"
        }
      }
    },
    "FlightStatus": {
      "type": "object",
      "required": [
        "FlightNumber"
      ],
      "properties": {
        "FlightNumber": {
          "type": "integer",
          "format": "int64"
        },
        "Airborne": {
          "type": "object",
          "properties": {}
        },
        "AircraftType": {
          "type": "string"
        },
        "FlightDelayDuration": {
          "type": "string"
        },
        "FlightArrivalDelayDuration": {
          "type": "string"
        },
        "AirportCode": {
          "type": "string"
        },
        "ScheduledDeparture": {
          "type": "string"
        },
        "ScheduledArrival": {
          "type": "string"
        }
      }
    },
    "errorModel": {
      "type": "object",
      "required": [
        "code",
        "message"
      ],
      "properties": {
        "code": {
          "type": "integer",
          "format": "int32"
        },
        "message": {
          "type": "string"
        },
        "Type": {
          "type": "string"
        },
        "Target": {
          "type": "string"
        },
        "Details": {
          "type": "string"
        },
        "InnerError": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/innerErrorModel"
          }
        }
      }
    },
    "innerErrorModel": {
      "type": "object",
      "required": [
        "code",
        "innererror"
      ],
      "properties": {
        "code": {
          "type": "integer",
          "format": "int32"
        },
        "innererror": {
          "type": "string"
        }
      }
    },
    "FlightInformation": {
      "type": "object",
      "properties": {
        "PALLlist": {
          "type": "string"
        },
        "SeatMap": {
          "type": "string"
        },
        "FlightEquipment": {
          "$ref": "#/definitions/FlightEquipment"
        }
      }
    },
    "FlightEquipment": {
      "type": "object",
      "properties": {
        "EquipmentType": {
          "type": "string"
        },
        "SeatMapSVG": {
          "type": "string"
        }
      }
    }
  }
}
```

**Comment by olydis** [comment]
# Problem
The reason for this is arguably unexpected: AutoRest does not try to be smart about status codes indicating success or failure. In your case, I see 200 and 400 response definitions, which will *both* be treated as "a valid way for the service to behave", hence the response schemas will *both influence the return type*. The "lowest common supertype" of your schemas is `object` (since they don't share another base).

# Solution
AutoRest treats `default` responses as the erroneous ones.
In other words, rename `400` response to `default` and everything should work as expected.
More detail on this topic: https://github.com/Azure/azure-rest-api-specs-pr/blob/master/documentation/creating-swagger.md#negative-responses
This solution may not be super satisfying since you are no longer explicit about the precise status code you expect in case of an error. We plan to improve the experience in the future (so you can somehow mark the `400` response as being erroneous)

**Comment by johnamcruz** [comment]
Thank you @olydis That worked for me.

**Comment by olydis** [comment]
Awesome, you're welcome! 👍

**Comment by Ricky-G** [comment]
anyone still looking at this, that link is broken, below will get you to the right place

https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#negative-responses
